tide_core.module

```diff
@@ -189,3 +191,45 @@ function tide_core_entity_bundle_create($entity_type_id, $bundle) {
     }
   }
 }
+
+/**
+ * Used by hook_ENTITY_TYPE_delete.
+ *
+ * When deleting a node randomly, we need to update the referencing node field
+ * and delete the Paragraph entity that the deleted node attached to.
+ */
+function _tide_core_clean_up_related_links($entity) {
+  // Instead of using Paragraph getParent function, we use
+  // database api directly, which is easy to read and understand.
+  $query = \Drupal::database()->select('node__field_related_links', 'n');
```

Comment:
@vincent-gao cc @anthony-malkoun @GROwen 
I'm extremely concerned about the performance impact of this solution. Upon any entity deletion, we join 2 large tables to find if anything references the to-be-deleted entity in *its link field*, then update them accordingly. Drupal is not supposed to behave that way.

IMO the error message from Drupal is correct as expected due to the link fields are configured to validate the destination links.

Reply:
I agree with @sonnykt - another solution could be to alter the default system message with something a little more helpful.

Reply:
Yes~~~. I agree. The error message is definitely correct and expected.
The validation logic is hidden under function`massageFormValues` from `contrib/paragraphs/src/Plugin/Field/FieldWidget/ParagraphsWidget.php`  `line 2274`.
there is no opportunity to alter this error message. or replaced the default widget(insane and no need).

Or

could be fixing by deleting the attached paragraph entity by using the entity query.